### PR TITLE
Fixing remaining stub mypy issues + run check-stubs to CI

### PR DESCRIPTION
**Pull request by sw23**
- Fixing a few remaining mypy complaints so that `make check-stubs` now passes
- Calling `make check-stubs` instead of `make stubs` in CI actions.
  - `make check-stubs` builds the stubs + runs mypy against them (returning a non-zero error code if errors pop up)

This is a follow-on PR from: #3621

I worked on this as part of #hacktoberfest (https://hacktoberfest.digitalocean.com/), so if the PR looks OK would you mind adding the label hacktoberfest-accepted?

### CODE REVIEW

File: shared-bindings/socketpool/SocketPool.c [hunk 1]
```diff
@@ -58,7 +58,7 @@ STATIC mp_obj_t socketpool_socketpool_make_new(const mp_obj_type_t *type, size_t
 }
 
 
-//|     def socket(self, family: int = AF_INET, type: int = SOCK_STREAM, proto: int = IPPROTO_TCP) -> None:
```
Comment (tannewt) [inline shared-bindings/socketpool/SocketPool.c:61]: Is there no way for this to include these defaults? They are helpful to have in docs.

**Review by tannewt — CHANGES_REQUESTED** [review]
One question but good otherwise!

File: shared-bindings/socketpool/SocketPool.c (hunk 1 above)
Comment (sw23) [reply shared-bindings/socketpool/SocketPool.c:61]: I went back and forth on this - they aren't currently defined in the python class.  I made it match the socket constructor:
```
//| class socket:
//|
//|     def __init__(self, family: int, type: int, proto: int) -> None:
//|         """Create a new socket
```
I'll update both of these to define them in the python class

Comment (sw23) [reply shared-bindings/socketpool/SocketPool.c:61]: Take a look now and let me know what you think - I added class attributes for the enum/define values

Comment (jepler) [reply shared-bindings/socketpool/SocketPool.c:61]: Is it deliberate that AF_INET is defined twice with different numeric values?

Comment (sw23) [reply shared-bindings/socketpool/SocketPool.c:61]: That is a good question - I matched what's in the C code, so the bigger question is does it make sense for the C header values to be the same?

`network/__init__.h`
```C
#define MOD_NETWORK_AF_INET (2)
#define MOD_NETWORK_AF_INET6 (10)

#define MOD_NETWORK_SOCK_STREAM (1)
#define MOD_NETWORK_SOCK_DGRAM (2)
#define MOD_NETWORK_SOCK_RAW (3)
```

`socketpool/SocketPool.h`
``` C
typedef enum {
    SOCKETPOOL_SOCK_STREAM,
    SOCKETPOOL_SOCK_DGRAM,
    SOCKETPOOL_SOCK_RAW
} socketpool_socketpool_sock_t;

typedef enum {
    SOCKETPOOL_AF_INET,
    SOCKETPOOL_AF_INET6
} socketpool_socketpool_addressfamily_t;
```

Comment (tannewt) [reply shared-bindings/socketpool/SocketPool.c:61]: Can we define them as `AF_INET = ...`? That way the implementation of the number stays an implementation detail.

Comment (sw23) [reply shared-bindings/socketpool/SocketPool.c:61]: Removed the implementation-specific values from the stubs

**Review by tannewt — APPROVED** [review]
Thank you!

**Pull request merged**